pb/sigchain.go

```diff
@@ -141,6 +145,56 @@ func (sc *SigChain) Verify() error {
 	return nil
 }
 
+func (sc *SigChain) VerifyMeta() error {
+	if sc.Length() < 3 {
+		return fmt.Errorf("sigchain should have at least 3 elements, but only has %d", sc.Length())
+	}
+
+	if len(sc.Elems[0].Id) > 0 && !bytes.Equal(sc.SrcId, sc.Elems[0].Id) {
+		return fmt.Errorf("sigchain has wrong src id")
+	}
+
+	if !sc.IsComplete() {
+		return fmt.Errorf("sigchain is not complete")
+	}
+
+	if bytes.Equal(sc.Elems[0].Id, sc.Elems[1].Id) {
+		return fmt.Errorf("sender and relayer 1 has the same ID")
+	}
+
+	if sc.IsComplete() && len(sc.Elems[sc.Length()-1].NextPubkey) > 0 {
+		return fmt.Errorf("next pubkey in last sigchain elem should be empty")
+	}
+
+	if sc.Elems[0].Mining {
+		return fmt.Errorf("first sigchain element should have mining set to false")
+	}
+
+	if sc.Elems[sc.Length()-1].Mining {
+		return fmt.Errorf("last sigchain element should have mining set to false")
+	}
+
+	for i, e := range sc.Elems {
+		if i == 0 || (sc.IsComplete() && i == sc.Length()-1) {
```

Comment:
The third time `sc.IsComplete()` checked.

Reply:
Removed duplicate check